### DISCUSSION TITLE
Built-in nodes VS Built-in extra nodes

### DISCUSSION THREAD

**Discussion by rhalaly** [comfyanonymous/ComfyUI#4542] [Q&A] [answered]
What is the difference between the built-in nodes placed in `nodes.py` and the built-in extra nodes placed in `comfy_extra` directory? Are they load differently?

**Comment by ltdrdata** [comment]
`comfy_extras` is like built-in custom nodes. If possible, avoid modifying `nodes.py` and create additional core nodes here instead.

**Comment by rhalaly** [comment]
What is the logic behind this decision to split them? Why not make all the "built-in" core nodes in the same place?

**Reply by ltdrdata** [reply]
The `nodes.py` file is already large enough. If we keep adding nodes to it, it will become completely unmanageable.

**Reply by rhalaly** [reply]
Yeah, I just don't understand why they maintain both, and not moving all the classes from `node.py` to dedicated nodes in `comfy_extras` and keep only the loading node logic in `node.py`. It feels much more natural and maintainable.

I believe that there is a concrete reason they decided to do that like it is now

**Reply by ltdrdata** [reply, marked as answer]
> Yeah, I just don't understand why they maintain both, and not moving all the classes from `node.py` to dedicated nodes in `comfy_extras` and keep only the loading node logic in `node.py`. It feels much more natural and maintainable.
> 
> I believe that there is a concrete reason they decided to do that like it is now

If we do that, all custom nodes that depend on `nodes.py` will collapse instantly.
The benefits of moving it are extremely minimal.

**Reply by rhalaly** [reply]
Ahh, so if I understand you correctly, you keep in `nodes.py` core nodes that are also can be reused in custom nodes, by inheritance, I guess.

If this is correct, this is what I missed
